### PR TITLE
Playwright: merge and migrate `Invite: Editor` and `Invite: Viewer` spec.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR proposes to combine and migrate the two remaining `Invite` specs from Selenium.

Key changes:
- new components, pages and methods to interact with underlying screens.
- streamline and restructure the resulting migrated spec.
- minor updates to JSDoc, selector naming and method signatures.

Background:

The two remaining `Invite` related specs - `Editor` and `Viewer` - have been merged into one resulting spec, as steps in the two specs overlapped significantly. In the migrated spec, only the user permission level for Editor is tested. 

By default, the test is skipped to reduce the number of new accounts being generated. To enable, we must first address #55600, as it is not necessary to run the signup spec with every commit.

Related to #55600

### CODE REVIEW

File: packages/calypso-e2e/src/lib/pages/people-page.ts [hunk 1]
```diff
@@ -34,11 +41,62 @@ export class PeoplePage {
...
+		const elementHandle = await this.page.waitForSelector(
+			selectors.deletedUserContentAction( 'delete' )
+		);
+
+		// Invoke scroll directly via JavaScript.
+		// Playwright's built-in `click` or `check` methods are not able to
+		// scroll the radio buttons into view for uncertain reasons.
+		await this.page.evaluate(
+			( element: SVGElement | HTMLElement ) => element.scrollIntoView(),
+			elementHandle
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/people-page.ts:87]: As noted in the comments, strangely Playwright isn't able to scroll this page - even with `elementHandle.scrollIntoViewIfNeeded`.

File: test/e2e/specs/specs-playwright/wp-invite__revoke.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ describe( DataHelper.createSuiteTitle( `Invite: Revoke` ), function () {
 		page = args.page;
 	} );
 
-	it( 'Can log in', async function () {
+	it( 'Log in', async function () {
```
Comment (worldomonation) [inline test/e2e/specs/specs-playwright/wp-invite__revoke.ts:30]: Removing modal verb.

File: test/e2e/specs/specs-playwright/wp-invite__revoke.ts [hunk 3]
```diff
@@ -52,7 +52,8 @@ describe( DataHelper.createSuiteTitle( `Invite: Revoke` ), function () {
 	it( 'Revoke the invite for test user', async function () {
 		await sidebarComponent.navigate( 'Users', 'All Users' );
 		await peoplePage.clickTab( 'Invites' );
-		await peoplePage.revokeInvite( testEmailAddress );
+		await peoplePage.selectInvitedUser( testEmailAddress );
+		await peoplePage.revokeInvite();
```
Comment (worldomonation) [inline test/e2e/specs/specs-playwright/wp-invite__revoke.ts:56]: Separated out the action to revoke invite and clicking on the pending invite into separate methods, so the actions are not coupled to one another.

File: test/e2e/specs/specs-playwright/wp-invite__new-user.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	InvitePeoplePage,
+	PeoplePage,
+	setupHooks,
+	UserSignupPage,
+	Roles,
+	BrowserManager,
+} from '@automattic/calypso-e2e';
+import { Page } from 'playwright';
+
+describe.skip( DataHelper.createSuiteTitle( `Invite: New User` ), function () {
```
Comment (worldomonation) [inline test/e2e/specs/specs-playwright/wp-invite__new-user.ts:15]: As noted in the description, the test is skipped by default, just as it was for Selenium.

This test is too expensive resource-wise to run every commit and would be better served running only once per trunk merge.

File: test/e2e/specs/specs-playwright/wp-invite__new-user.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+				role: role as Roles,
+				message: `Test invite for role of ${ role }`,
+			} );
+		} );
+
+		it( 'Confirm invite is pending', async function () {
+			await sidebarComponent.navigate( 'Users', 'All Users' );
+			await peoplePage.clickTab( 'Invites' );
+			await peoplePage.selectInvitedUser( email );
+			throw new Error();
```
Comment (bsessions85) [inline test/e2e/specs/specs-playwright/wp-invite__new-user.ts]: What is this for?

File: test/e2e/specs/specs-playwright/wp-invite__new-user.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+		} );
+
+		it( 'View invited user in Team', async function () {
+			peoplePage = new PeoplePage( page );
+			await peoplePage.selectUser( username );
+		} );
+
+		it( 'Remove invited user from site', async function () {
+			await peoplePage.deleteUser();
+		} );
```
Comment (bsessions85) [inline test/e2e/specs/specs-playwright/wp-invite__new-user.ts:126]: Is there a way to verify that the user has actually been deleted?

**Review by bsessions85 — COMMENTED** [review]
This is looking good, just left a couple questions. Confirmed that test passes locally, as long as I remove `throw new Error();`

File: test/e2e/specs/specs-playwright/wp-invite__new-user.ts (hunk 6 above)
Comment (bsessions85) [reply test/e2e/specs/specs-playwright/wp-invite__new-user.ts:126]: Nevermind, I missed that you were looking for the banner in the `deleteUser` function

Comment (worldomonation) [reply test/e2e/specs/specs-playwright/wp-invite__new-user.ts:126]: Yeah, I abstracted away the check into the method itself. I did make the suggestion (forgot to who) to not let methods do too many things at once and technically this violates that, so...

I still struggle with it :)

File: test/e2e/specs/specs-playwright/wp-invite__new-user.ts (hunk 5 above)
Comment (worldomonation) [reply test/e2e/specs/specs-playwright/wp-invite__new-user.ts]: D'oh, that was for testing and it slipped through. Thanks for catching it!

Comment (worldomonation) [reply test/e2e/specs/specs-playwright/wp-invite__new-user.ts]: Fixed in 94346b774539cfded29062bf6f81f197c573387c.

**Review by bsessions85 — APPROVED** [review]
LGTM 🚢

**Pull request merged**